src/Plutarch/Extra/IsData.hs

```diff
@@ -105,11 +118,19 @@ gProductFromBuiltinData (BuiltinData (List xs)) = do
     productTypeTo <$> hctraverse (Proxy @FromData) (unI . mapKI fromData) prod
 gProductFromBuiltinData _ = Nothing
 
-{- |
-  Unsafe version of 'gProductFromBuiltinData'.
+-- | @since 1.4.0
+newtype DerivePDataFieldsViaDataList (a :: S -> Type) (s :: S)
+    = DerivePDataFieldsViaDataList (a s)
 
-  @since 1.1.0
--}
+-- | @since 1.4.0
+instance
+    (PDataFields (PInner a), PInner (DerivePDataFieldsViaDataList a) ~ a) =>
```

Comment:
PInner no longer has the awkward extra type variable :partying_face: !

Reply:
Yeah that's a huge improvement:)